src/app/users/[username]/page.tsx

```diff
@@ -11,40 +17,104 @@ type PropTypes = {
 }
 
 export default async function User({ params }: PropTypes) {
-    const { user, permissions, memberships } = await getUser({
+    const { user } = await getUser({
         userRequired: true,
         shouldRedirect: true,
         returnUrl: `/users/${params.username}`,
     })
 
-    const me = params.username === 'me'
+    const me = params.username === 'me' || params.username === user.username
     const username = me ? user.username : params.username
 
-    // TODO REFACTOR
-    const userProfile = await prisma.user.findUnique({
+    const profile = await readUserProfile({username})
+
+    if (!profile) {
+        notFound()
+    }
+
+    const groupIds = profile.memberships.map(group => group.groupId)
+    const committees = await readCommitteesFromIds(groupIds)
+
+    const studyProgramme = await prismaCall(() => prisma.studyProgramme.findFirst({
```

Comment:
This should be an action. If you think we can refactor this at a later date add a todo comment.

Reply:
As I mentioned I will change this on the visibility branch. There I have made functions to infer information about users and groups. The way I infer a users order is by looking at their membership order to the omega membership group, not study program. I think this makes more sense as if you do not have a membership to the omega member group you do not even have a order and your profile should say soelle or external